fix: better ordering for exact search matches

rkaiser0324 opened the pull request:
Previously, the logic would not capture exact matches (e.g., if you search for "woocommerce" you would get 130 pages of results but it doesn't look like it actually includes the exact match).  This is because it was doing a LIKE on "%woocommerce%" so not including the term itself.

Now, we explicitly check for that term in the beginning, middle, end of the results as well as for an exact match.  In addition, we have a smarter ordering where any exact match is at least listed first.  So now you don't have to scroll through hundreds of pages.

Also, drop the apparently-obsolete "active_only" parameter from search as it wasn't used anymore.

NoelLH commented:
Hey @rkaiser0324, thanks for the contribution!

I definitely get the frustration here and like the idea + appreciate you contributing, but I think there are a couple of general points before we get into reviewing the specifics:

1. I'm not sure about the active flag being _apparently-obsolete_ – if you check `Update::deactivate()` and its invocations you can see places where we do deactivated packages for various reasons. It might not match your intuition but I don't think we should rule out the possibility that somebody is using this distinction, and in any case it sounds like it would be better handled distinctly on a separate discussion / PR.
2. `%` includes zero characters. So I don't think we need all the new conditions, and I'm worried they could hurt query performance.

rkaiser0324 commented:
@NoelLH -

1. When I said "apparently obsolete" I was just referring to the `active_only` request parameter on `/search/`.  That had already been commented out of `searchbar.twig` (the only other place it was referenced in the codebase), so I removed the sort and filter logic for this parameter from `MainController.php`.  All your logic for active/inactive packages is of course still there. 
2. Point taken.  I reverted the query logic to use the two `LIKE` conditions as before. The important part of this PR is really the improved sort logic for exact matches.

NoelLH commented:
@rkaiser0324 Thanks! I misread what was happening with the active field support removal – got you now.

rkaiser0324 commented:
You're right on the injection risk.  I dropped the literals and added a second bound parameter.

NoelLH commented:
Thanks @rkaiser0324! Merged
